File: t/VB/MedicalViewerLayoutDemo/UI/RemoveCellDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace MedicalViewerLayoutDemo
   Public Partial Class RemoveCellDialog : Inherits Form
	  Public Sub New(ByVal owner As MainForm)
		 InitializeComponent()

		 ' set the allowed range for the cell index text box.
		 cellIndexText.MinimumAllowed = -1
		 cellIndexText.MaximumAllowed = owner.Viewer.Cells.Count - 1
            cellIndexText.Text = (0).ToString()
            removeSelectedRadio.Enabled = owner.GetSelectedCell() IsNot Nothing
            If removeSelectedRadio.Enabled Then
                removeSelectedRadio.Checked = True
            End If

	  End Sub

	  Private Sub removeSpecificRadio_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles removeSpecificRadio.CheckedChanged
		 cellIndexLabel.Enabled = removeSpecificRadio.Checked
		 cellIndexText.Enabled = removeSpecificRadio.Checked
		 If removeSpecificRadio.Checked Then
			cellIndexText.Focus()
		 End If
	  End Sub

	  Private Sub okButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnOK.Click
		 Try
			If removeSpecificRadio.Checked Then
			   CType(Me.Owner, MainForm).Viewer.Cells.RemoveAt(Convert.ToInt32(cellIndexText.Text))
			ElseIf removeSelectedRadio.Checked Then
			   CType(Me.Owner, MainForm).RemoveSelectedCells()
			ElseIf removeAllRadio.Checked Then
			   CType(Me.Owner, MainForm).Viewer.Cells.Clear()
			End If
		 Catch ex As Exception
			Messager.ShowError(Me, ex)
		 End Try
		 Me.Close()
	  End Sub
   End Class
End Namespace

File: t/CS/MedicalViewerDemo/UI/RemoveCellDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools.Demos;

namespace MedicalViewerDemo
{
   public partial class RemoveCellDialog : Form
   {
      public RemoveCellDialog(MainForm owner)
      {
         InitializeComponent();

         // set the allowed range for the cell index text box.
         cellIndexText.MinimumAllowed = -1;
         cellIndexText.MaximumAllowed = owner.Viewer.Cells.Count - 1;
         cellIndexText.Text = (0).ToString();
      }

      private void removeSpecificRadio_CheckedChanged(object sender, EventArgs e)
      {
         cellIndexLabel.Enabled = removeSpecificRadio.Checked;
         cellIndexText.Enabled = removeSpecificRadio.Checked;
         if (removeSpecificRadio.Checked)
            cellIndexText.Focus();
      }

      private void okButton_Click(object sender, EventArgs e)
      {
         try
         {
            if (removeSpecificRadio.Checked)
               ((MainForm)this.Owner).Viewer.Cells.RemoveAt(Convert.ToInt32(cellIndexText.Text));
            else if (removeSelectedRadio.Checked)
            {
               ((MainForm)this.Owner).RemoveSelectedCells();
            }
            else if (removeAllRadio.Checked)
            {
               ((MainForm)this.Owner).Viewer.Cells.Clear();
            }
         }
         catch (Exception ex)
         {
            Messager.ShowError(this, ex);
         }
         this.Close();
      }
   }
}
